docs/tests-mvp.md

```diff
@@ -0,0 +1,265 @@
+- (Minimize) Time spent pinning/unpinning & garbage collection
+- (Minimize) Number of unreferanceable nodes when adding multiple files to MF
+- (Minimize) Waste when adding multiple files to an MFS tree (nodes that are no longer referenceable from the MFS tree due to graph updates)
+- (Minimize) Time spent creating the Manifest files uisng URL and File Store
+
+#### Execution Variants
+
+- Ran with with an arbitraty amount of nodes (from 10 to 1000000) - N
+
+#### Test Narrative
+
+- **Warm up**
+  - Each node accesses a pre-generated file test set (fixtures) that contains:
+    - Small files (1KB to 100MB)
+    - Medium files (100MB to 10GB)
+    - Large files (10GB to 10TB)
+    - Nested directories with multiple levels (10 to 100000) containing small files
+- **Wave I** - IPFS Add with multiple layouts and sharding
+  - Each node adds each file using Balanced Dag & Trickle Dag
+  - Each node adds the nested directories using regular add and sharding
+- **Wave II** - MFS
+  - Each node calls files.write to create the Nested Directories structure in MFS. Once regular and once with Sharding
+  - Each node runs a pin on the MFS root hash
+  - Each node runs GC (to clean all the waste created by MFS)
+- **Wave III** - URL Store & File Store
+  - The Nested Directories get served beyond a HTTP static server
+  - Each node adds (creates the manifest) for the nested directories using URL Store & File Store
+
+### 2. Data Transfer of Random DataSets (Bitswap/GraphSync)
+
+Create an envinroment in which data transfer is stress tested. This test is not about content discovery or connectivity, it is assumed that all nodes are dialable by each other and that these are executed in an homogeneus network (same CPU, Memory, Bandwidth).
+
+#### What is being optimized (min/max, reach)
+
+- (Minimize) The performance of fetching a file. Lower is Better
+  - To compute this, capture: file size, number of nodes in the IPLD graph, time to fetch it from first block to completion.
+- (Minimize) The bandwidth consumed to fetch a file. Lower is Better
+  - To compute this, capture: The number of duplicated blocks received vs. total block count
+- (Reach) The number of nodes that were able to fetch all files as instructed. (Reach 100% success of all fetches)
+- (Reach) No node is expected to crash/panic during this Test Plan. (Reach 0% crashes)
+
+#### Execution Variants
+
+This test is complete if one can:
+
+- Ran with with an arbitraty amount of nodes (from 10 to 1000000) - N
+- Ran with custom libp2p & IPFS suites (swap in/out Bitswap & GraphSync versions, Crypto Channels, Transports and other libp2p components)
+
+This test is not expected to support:
+
+- An heterogeneus network in which nodes have different configurations
+
+#### Test Narrative
+
+- **Warm up**
+  - Boot N nodes
+  - Connect each node to the node next to it (hash ring)
+  - Run multiple DHT random-walk queries to populate the finger tables
```

Comment:
Feels like we're mixing tests here that is bound to cause problems? Any reason not to test the DHT separately and manually control node connections for the Data Transfer tests?

Reply:
@aschmahmann this step is specifically in the Warm up phase to "set up the network into the testing state" (see description at the top of the document), it is not testing the DHT, however, it is important to have a well connected fabric to be able to test the rest. There will be another test that focuses on the DHT connectivity.

Reply:
>There will be another test that focuses on the DHT connectivity

👍 

> it is not testing the DHT

Maybe not fully, but if the DHT is not working to some specification then the test will not perform as it should. We may be confident that the DHT currently works enough to satisfy this specification, but it doesn't mean that we haven't coupled the tests. Given that we already `Connect each node to the node next to it (hash ring)` shouldn't we just manually populate the finger tables with the appropriate random distribution?